packages/datetime/src/dateInput.tsx

```diff
@@ -98,6 +98,12 @@ export interface IDateInputProps extends IDatePickerBaseProps, IProps {
     outOfRangeMessage?: string;
 
     /**
+     * Whether the date popover should be inline.
+     * @default true
+     */
+    popoverInline?: boolean;
```

Comment:
+1 to everything there. i'm also quite certain there will be focus issues with non-inline popovers.